### ISSUE TITLE
Blank/empty HTML comments result in loss of information in that element

### ISSUE THREAD

**Issue by dgtlmoon** [weblyzard/inscriptis#45]
version 2.2.0

```
from inscriptis import get_text

x = get_text('<html><body><span class="price-detailed__unit-price"><span>$<!-- -->90<!-- -->.<!-- -->74</span></span></body></html>')
print (x)
# $

x = get_text('<html><body><span class="price-detailed__unit-price"><span>$90.74</span></span></body></html>')
print (x)
# 90.74
```
I have no idea why this is the case :/ I noticed some sites may use this to stop scrapers

**Comment by dgtlmoon** [comment]
For now I'm just regexing this out https://github.com/dgtlmoon/changedetection.io/pull/764/files#diff-97e35d8550009a723110a24b826e4329175e4d159ebe1f2cca6b684fc45197c5R216

**Comment by AlbertWeichselbraun** [comment]
thank you for reporting this issue.

i did a first analysis and see two ways of addressing this:

 1. filter out comments with regexps (very easy to do for the general case, but tricky if we want to consider more sophisticated settings such as comments in pre-text, nested comments, etc.).
 2. parse comment nodes for text (this seems to be the sustainable solution).

I will add your example to the unittests cases and do some more additional testing before commiting to a solution.

**Comment by dgtlmoon** [comment]
@AlbertWeichselbraun anytime :) Thanks for the great project!

Here's an example in the wild https://www.homedepot.com/p/Home-Decorators-Collection-Hand-Scraped-Strand-Woven-Sahara-3-8-in-T-x-5-1-5-in-W-x-36-02-in-L-Engineered-Click-Bamboo-Flooring-HL655H/300011064

**Comment by AlbertWeichselbraun** [comment]
I have created a fix that draws upon parsing node comments (i.e., should be considerably more robust than the regex) in the branch `fix/html-comment-ofuscation`
  https://github.com/weblyzard/inscriptis/tree/fix/html-comment-ofuscation

may I ask you to check, whether this solves the problem (so far I can confirm that it works with both examples you provided).

Many thanks, Albert

**Comment by dgtlmoon** [comment]
@AlbertWeichselbraun apologies, i only just saw that! I cant find that branch

**Comment by dgtlmoon** [comment]
with version `2.4.0.1` there is a whitespace added
```
>>> from inscriptis import get_text
>>> 
>>> x = get_text('<html><body><span class="price-detailed__unit-price"><span>$<!-- -->90<!-- -->.<!-- -->74</span></span></body></html>')
>>> print(x)
$ 90.74
```

**Comment by dgtlmoon** [comment]
Maybe i'm missing some setting that enforces a whitespace where there is a comment block?

```
>>> x = get_text('<html><body><span class="price-detailed__unit-price"><span>$90.<!-- silent -->74</span></span></body></html>')
>>> x
'$90. 74'
```

**Comment by AlbertWeichselbraun** [comment]
Using strict indentation handling fixes this (see #83).

**Comment by dgtlmoon** [comment]
@AlbertWeichselbraun thanks so much :)
